dashboard/lib/school_info_deduplicator.rb

```diff
@@ -20,8 +20,24 @@ def process_school_info_attributes(school_info_attr)
     attr = school_info_attr.symbolize_keys
 
     # Names of state and zip fields change between form fields and SchoolInfo class
-    attr[:state] ||= school_info_attr['school_state']
-    attr[:zip] ||= school_info_attr['school_zip']
+    attr[:state] ||= school_info_attr[:school_state]
+    attr[:zip] ||= school_info_attr[:school_zip]
+
+    # Remove empty attributes.  Notably school_district_id can come through
+    # as an empty string when we don't want anything.
+    attr.delete_if {|_, e| e.blank?}
```

Comment:
Will this still have issues where, for instance, one record has (a=1, b=2, c=3) and we try to look up (a=1, b="", c="") and the matching will end up doing a lookup with where({a=1}) instead of where({a=1, b="", c=""})? i.e. potentially mistakenly matching against an existing record, whereas what we really want is a new one

Reply:
I don't believe so, you can double check the unit tests that I have added to see, input vs output of the method. It's actually doing this already, I just moved it up before so that I can add the nil filters on school_district_id and school_id. I moved them above the slice, because in addition to slice removing properties that are not in the provided list, it also re-orders them and it seemed friendlier for debugging to have a consistent order.

Reply:
Yes, definitely, at the minimum, it will produce better matching that what we have currently.